ctsm5.3.050: FUNITCTSM tests fail on Izumi due to a ccs_config update

### Brief summary of bug

The FUNITCTSM tests fail on Izumi due to an update for ccs_config in ccs_config_cesm1.0.42

### General bug information

**CTSM version you are using:** ctsm5.3.045-181-g54e640b6

**Does this bug cause significantly incorrect results in the model's science?** [Yes / No]

**Configurations affected:** Just the PF unit testing and only on Izumi

### Details of bug

FUNITCTSM_P1x1.f10_f10_mg37.I2000Clm50Sp.izumi_intel

fails at build time (as well as when run by hand under the src directory).

This starts with ccs_config_cesm1.0.42

### Important output or errors that show the problem

The error on build looks like this:

```
 ... Done.  Results in test_filterBuilder.F90
 ... Done.  Results in test_unittestArray.F90
 ... Done.  Results in test_quadratic.F90
 ... Done.  Results in test_unittestSubgrid.F90
 ... Done.  Results in test_annual_flux_dribbler.F90
 ... Done.  Results in test_find_k_max_indices.F90
 ... Done.  Results in test_truncate_small_values.F90
 ... Done.  Results in test_clm_time_manager.F90
[ 45%] Generating test_convert_to_logical.F90
[ 49%] Building Fortran object clm_utils_test/annual_flux_dribbler_test/CMakeFiles/annual_flux_dribbler.dir/annual_flux_dribbler_driver.F90.o
[ 50%] Building Fortran object clm_unit_test_shr_test/unittestSubgrid_test/CMakeFiles/unittestSubgrid.dir/test_unittestSubgrid.F90.o
[ 50%] Building Fortran object clm_unit_test_shr_test/unittestFilterBuilder_test/CMakeFiles/unittestFilterBuilder.dir/test_filterBuilder.F90.o
[ 50%] Building Fortran object clm_utils_test/clm_time_manager_test/CMakeFiles/clm_time_manager.dir/clm_time_manager_driver.F90.o
[ 50%] Building Fortran object clm_utils_test/quadratic_test/CMakeFiles/quadratic.dir/quadratic_driver.F90.o
[ 50%] Building Fortran object clm_unit_test_shr_test/unittestArray_test/CMakeFiles/unittestArray.dir/test_unittestArray.F90.o
[ 50%] Building Fortran object clm_utils_test/numerics_test/CMakeFiles/numerics.dir/numerics_driver.F90.o
Processing file ../../../../../../../../../fs/cgd/data0/erik/ctsm_worktree/externals_update/src/utils/test/array_utils_test/test_convert_to_logical.pf
 ... Done.  Results in test_convert_to_logical.F90
[ 50%] Building Fortran object clm_utils_test/array_utils_test/CMakeFiles/array_utils.dir/array_utils_driver.F90.o
[ 50%] Building Fortran object clm_utils_test/quadratic_test/CMakeFiles/quadratic.dir/test_quadratic.F90.o
[ 50%] Building Fortran object clm_utils_test/annual_flux_dribbler_test/CMakeFiles/annual_flux_dribbler.dir/test_annual_flux_dribbler.F90.o
[ 50%] Building Fortran object clm_utils_test/numerics_test/CMakeFiles/numerics.dir/test_truncate_small_values.F90.o
[ 51%] Building Fortran object clm_utils_test/clm_time_manager_test/CMakeFiles/clm_time_manager.dir/test_clm_time_manager.F90.o
[ 51%] Building Fortran object clm_utils_test/array_utils_test/CMakeFiles/array_utils.dir/test_convert_to_logical.F90.o
/scratch/cluster/erik/FUNITCTSM_P1x1.f10_f10_mg37.I2000Clm50Sp.izumi_intel.20250528_122148_6b5ugm/bld/__command_line_test__/__command_line_test__/clm_utils_test/annual_flux_dribbler_test//fs/cgd/data0/erik/ctsm_worktree/externals_update/src/utils/test/annual_flux_dribbler_test/test_annual_flux_dribbler.pf(133): warning #5462: Global name too long, shortened from: test_annual_flux_dribblerdeltaaddedmidyear_ignoredinlatertimesteps_mp_DELTA_SHOULD_BE_IGNORED to: st_annual_flux_dribblerdeltaaddedmidyear_ignoredinlatertimesteps_mp_DELTA_SHOULD_BE_IGNORED
    real(r8), parameter :: delta_should_be_ignored(1) = [3.e6_r8]
---------------------------^
[ 51%] Building Fortran object clm_unit_test_shr_test/unittestFilterBuilder_test/CMakeFiles/unittestFilterBuilder.dir/unittestFilterBuilder_driver.F90.o
[ 51%] Linking Fortran executable unittestFilterBuilder
ifort: error #10236: File not found:  '$(NETCDF_PATH)/lib/libnetcdff.a'
make[2]: *** [clm_unit_test_shr_test/unittestFilterBuilder_test/CMakeFiles/unittestFilterBuilder.dir/build.make:121: clm_unit_test_shr_test/unittestFilterBuilder_test/unittestFilterBuilder] Error 1
make[1]: *** [CMakeFiles/Makefile2:1898: clm_unit_test_shr_test/unittestFilterBuilder_test/CMakeFiles/unittestFilterBuilder.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.10)
 
 list(APPEND CMAKE_MODULE_PATH ${CIME_CMAKE_MODULE_DIRECTORY})
 include(CIME_initial_setup)
@@ -93,7 +93,7 @@ add_library(csm_share ${share_sources} ${drv_sources_needed})
 declare_generated_dependencies(csm_share "${share_genf90_sources}")
 add_library(clm ${clm_sources})
 declare_generated_dependencies(clm "${clm_genf90_sources}")
-add_dependencies(clm csm_share esmf)
+add_dependencies(clm csm_share ESMF)
 
 # We need to look for header files here, in order to pick up shr_assert.h
 include_directories(${CLM_ROOT}/share/include)

